src/client_side_request.cc

```diff
@@ -1662,7 +1662,7 @@ ClientHttpRequest::checkForInternalAccess()
     if (!internalCheck(request->url.path()))
         return;
 
-    if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
+    if (internalHostnameIs(SBuf(request->url.host())) && request->url.port() == getMyPort()) {
```

Comment:
Please reorder to reduce the number of cases applicable to this PR-added performance regression:

```suggestion
    if (request->url.port() == getMyPort() && internalHostnameIs(SBuf(request->url.host()))) {
```

Alternatively, you can leave internalHostnameIs() parameter type unchanged in this PR and continue to compare with a c-string until Uri::host() itself is converted to SBuf. Your call.